primitives/runtime-interface/src/pass_by.rs

```diff
@@ -417,17 +417,17 @@ impl<T: Copy + Into<u8> + TryFrom<u8, Error = ()>> PassByImpl<T> for Enum<T> {
 
 	fn into_ffi_value(instance: &T) -> WrappedFFIValue<Self::FFIType, Self::Owned> {
 		let value: u8 = (*instance).into();
-		value.into()
+		(value as u32).into()
 	}
 
 	fn from_ffi_value(arg: Self::FFIType) -> T {
-		T::try_from(arg).expect("Host to wasm provides a valid enum discriminant; qed")
+		T::try_from(arg as u8).expect("Host to wasm provides a valid enum discriminant; qed")
 	}
 }
 
-/// The type is passed as `u8`.
+/// The type is passed as `u32`.
 ///
 /// The value is corresponds to the discriminant of the variant.
 impl<T: Copy + Into<u8> + TryFrom<u8>> RIType for Enum<T> {
-	type FFIType = u8;
+	type FFIType = u32;
```

Comment:
Yes. WASM FFI doesn't support 8-bit and 16-bit integers natively, so it was being converted into an `u32` later anyway. So I've put the extra `WasmTy` bound on the `FFIType` so that it will be guaranteed that's actually FFI-compatible and moved the conversion into the native WASM FFI type earlier.